### PR TITLE
MODORDSTOR-22 - Implement GET /orders w/ search and filtering

### PR DESCRIPTION
**Pull request by khandramai**
[MODORDSTOR-22](https://issues.folio.org/browse/MODORDSTOR-22)

## Purpose
In order to support search and filtering across the various orders tables, we'll need to create a view and GET /orders API is for querying that view. 

## Approach
Existing GET /orders API refactored to allow search and filtering.

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

### CODE REVIEW

**Comment by craigmcnally** [comment]
I've accepted this as everything looks good and coverage is acceptable.  However, I'll add that it would be nice to have coverage on at least some of the exceptional code paths...

**Comment by khandramai** [comment]
@craigmcnally,
you are right - it seems not so easy to cover these exceptional paths of HelperUtils during Orders/Receiving_History unit testing. But I can propose to cover generic static methods from HelperUtils by separate test exactly for this class, probably, with using mock technique. Let me know if we need this now and I can provide some solution in scope of current PR.

File: src/main/java/org/folio/rest/persist/QueryHolder.java [hunk 1]
```diff
@@ -9,13 +9,15 @@
 public class QueryHolder {
 
   private String table;
+  private String searchField;
   private String query;
   private int offset;
   private int limit;
   private String lang;
 
-  public QueryHolder(String table, String query, int offset, int limit, String lang) {
```
Comment (piotr-kalashuk) [inline src/main/java/org/folio/rest/persist/QueryHolder.java]: You could leave constructor with default behavior to not modify any other classes:
```java
  public QueryHolder(String table, String query, int offset, int limit, String lang) {
    this(table, HelperUtils.JSONB, query, offset, limit, lang);
  }
```

**Comment by craigmcnally** [comment]
@khandramai I'll let you decide... if you think it's enough work to warrant a separate story, go ahead and create one or let me know and I will.  If you think you can squeeze it into this story w/o much effort even better.

**Comment by khandramai** [comment]
Work on HelperUtils test coverage will be continued in separate PR.

**Pull request merged**